### PR TITLE
add v1beta draft for designs

### PR DESCRIPTION
**Pull request by aabidsofi19**
**Notes for Reviewers**

This PR is a proposal for v1beta1 for designs . some parts might not be technically correct !! 

## Motivation

The current design schema, although undocumented, is a mix of our internal model constructs, deprecated Open Application Model (OAM) elements, and Cytoscape objects. This inconsistency makes it difficult to manage and analyze data effectively.

This revision proposes a stricter schema based on our well-defined internal constructs like models, components, and relationships.

**Example of Current Design (replace with an actual example)**

Much of the current content, particularly traits, lacks a defined schema and acts as a repository for Cytoscape-specific information.

## New in V1 Beta 1

**Removal of Services and Traits:**

The revamped component schema in V1 Beta 1 enhances component versatility for representing infrastructure entities, annotations, and presentation. This allows us to remove the need for both services and traits.

**Design-Level Control:**

V1 Beta 1 introduces the ability to store design-level preferences, including specifying permitted relationships between components.

**Explicit Relationships:**

The current schema relies on implicit Cytoscape constructs to define relationships. V1 Beta 1 introduces a dedicated "relationships stanza" for explicitly defining connections. This stanza uses a structured format with attributes like `<from>`, `<to>`, `<kind>`, `<type>`, `<subtype>`, and `<status>` to improve clarity.

## Versioning of Design

Designs now have version information stored as an integer (`int`) independent of the specific versioning system used. This means the version number itself is an integer, and the logic for assigning and incrementing it can differ depending on the chosen system.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: schemas/constructs/v1beta2-draft/designs-v1beta1.json [hunk 1]
```diff
@@ -0,0 +1,102 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Design Schema",
+  "description": "Schema for design  in v1Beta1",
```
Comment (leecalcote) [inline schemas/constructs/v1beta2-draft/designs-v1beta1.json:4]:
```suggestion
  "description": "Designs are your primary tool for collaborative authorship of your infrastructure, workflow, and processes.",

```

**Comment by leecalcote** [comment]
Will the concept of Designs and how they fit into a Catalog continue to be separately considered?

- Metadata in consideration of Designs as content in a Catalog; e.g. support level, technology, type, etc..
- Author, Owner, Created At, Updated At, etc.

Should Designs self-contain their own summarization of statistics like:

- Number of components
   - annotation and non-annotation components
- Number of relationships

**Comment by dragon-slayer875** [comment]
@aabidsofi19 Will it be a good question to ask to what level of conversion will be required to map this schema to the RJSF format?
Or at what point will it precisely occur?

**Comment by leecalcote** [comment]
Whoa. The stalebot showed up here.

File: schemas/constructs/v1beta2-draft/designs-v1beta1.json [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      "type": "string",
+      "description": "Name of the design"
+    },
+    "schemaVersion": {
+      "$ref": "https://schemas.meshery.io/v1alpha2/core.json#versionString",
+      "description": "Specifies the version of the schema to which the design  conforms."
+    },
+    "version": {
+      "$ref": "https://schemas.meshery.io/v1alpha2/core.json#semverString",
+      "description": "Version of the design",
```
Comment (leecalcote) [inline schemas/constructs/v1beta2-draft/designs-v1beta1.json:17]:
```suggestion
      "description": "Revision of the design as expressed by an auto-incremented, SemVer-compliant version number. May be manually set by a user or third-party system, but will always be required to be of version number higher than the previously defined version number.",

```

**Comment by MUzairS15** [comment]
I am unable to push on your branch, I will merge this PR and open other PR with updates.

**Pull request merged**